### DISCUSSION TITLE
Run xUnit using GitHub Action in a .NetFramework project.

### DISCUSSION THREAD

**Discussion by sonomirco** [xunit/xunit#2653] [Question] [answered]
Hi there, 
Could you please help me with this task?
I tried setting a target, as explained here
https://xunit.net/docs/getting-started/netfx/cmdline

But I need to add it to all the other projects I have in the solution, and after I tried, this is what I got.
![image](https://user-images.githubusercontent.com/57708659/213977846-324fcf71-1483-49bd-b253-9435a9250ac6.png)

Because I am using.
![image](https://user-images.githubusercontent.com/57708659/213977957-de0e1b0d-ad04-453a-98b5-e2bc073b4d75.png)

Could it be a solution using this action?
https://github.com/microsoft/vstest-action

The process will be as follows:
1. restore the package
2. build the solution
3. use `vstest-action` with the specific test assembly.

Any other suggestions are welcome.
Thanks

**Comment by bradwilson** [comment, marked as answer]
I don't know what `MSBuild.AssemblyVersion` is; it definitely has nothing to do with xUnit.net.

We don't (or recommend) use any custom actions to run our tests inside GitHub Actions. You should just be able to run the tests the same way you would run them on your local machine: either with `dotnet test` or by invoking our console runner.

There's very little help that can be given without seeing a repro project. The error messages here have nothing to do with xUnit.net. Perhaps you can provide a repro?

**Reply by bradwilson** [reply]
Also, the snippet you're showing makes me think you're using old-style (non-SDK) project files. While these _might_ work, they're also many years out of date and very well may be the cause of the build errors you're seeing.

**Reply by sonomirco** [reply]
@bradwilson MSBuild.AssemblyVersion is something I use to pass the versions that needs to be used when I build using msbuild.exe.

This link shows a solution which seems working.

https://dev.to/felipetofoli/GitHub-actions-for-net-full-framework-build-and-test-299h

Unfortunately, I can’t share the repo because it is private.
If there is a better way than what is shown in this post to run the tests, I will follow your suggestions.

**Reply by sonomirco** [reply]
I used the process shown in the post, adding the creation report at the end and it works.

**Reply by bradwilson** [reply]
That link is a 404 for me.

**Reply by sonomirco** [reply]
Could you try this one?
https://dev.to/felipetofoli/github-actions-for-net-full-framework-build-and-test-299h

**Reply by bradwilson** [reply]
Sample project: https://github.com/bradwilson/xunit-2653/

"Successful" Actions run: https://github.com/bradwilson/xunit-2653/actions/runs/4019577161/jobs/6906585606

(I put successful in quotes, because I explicitly sent in a failing test, because I wanted to see the tests run with 1 passing and 1 failing)

I'm still on the belief that whatever your problem is, it has nothing to do with us.

**Reply by sonomirco** [reply]
Yep, this is what I have done as well.
I was not pointing out a problem about xUnit; I raised a question showing what I did and the problems I got.
I think the title of this thread is relatively clear `Run xUnit using GitHub Action in a .NetFramework project. `I never thought there was a problem with xUnit.

Thanks so much for your time and for testing the solution. 👍
